### PR TITLE
New method added to stream json upserts to Soda. 

### PR DESCRIPTION
**Pull request by AEscobarCruz**
This will be helpful for very large upserts and in particular help geospace not have to keep too much in memory during uspert time.

### CODE REVIEW

File: soda-fountain-external/src/main/scala/com/socrata/soda/external/SodaFountainClient.scala [hunk 1]
```diff
@@ -70,7 +70,22 @@ class SodaFountainClient(httpClient: HttpClient,
...
+  /**
+   * Sends a request to Soda Fountain to upsert rows to a dataset
+   * and returns the response
+   * @param resourceName Resource name of the dataset to upsert to
+   * @param jIterator JValue iterator to stream
+   * @return HTTP response code and body
+   */
+  def upsertStream(resourceName: String, jIterator: Iterator[JValue]): Result = {
+    val upsertIterator = Iterator.single(StartOfArrayEvent()(Position.Invalid)) ++
+                          jIterator.flatMap {item=> JValueEventIterator(item) } ++
```
Comment (velvia) [inline soda-fountain-external/src/main/scala/com/socrata/soda/external/SodaFountainClient.scala]: (nit) spacing around item

**Comment by velvia** [comment]
LGTM other than one minor spacing thing

**Comment by velvia** [comment]
Cool thanks.  @AEscobarCruz how urgent is this change?

**Comment by AEscobarCruz** [comment]
No urgency from my end doing making these changes on my "free time", and wouldn't mind having others testing this further.

**Comment by urmilan** [comment]
:+1: Nice @AEscobarCruz !

**Comment by velvia** [comment]
jenkins ok to test

**Pull request merged**